File: code/ch20/20.2.1. 1291.sequentialDigits.py
class Solution:
    def sequentialDigits(self, low: int, high: int) -> List[int]:
        numbers = "123456789"
        ins = []
        n = len(numbers)
        for i in range(1, n):
            for j in range(n - i):
                ins.append(int(numbers[j : i + j + 1]))
        return [x for x in ins if x >= low and x <= high]

File: code/ch20/20.2.1. 1291.sequentialDigits.cpp
#include <vector>
#include <string>
using namespace std;

class Solution {
public:
  vector<int> sequentialDigits(int low, int high) {
    string numbers("123456789");
    vector<int> ins;
    int n = numbers.size();
    for (int i = 1; i < n; i++)
      for (int j = 0; j < n - i; j++)
        ins.push_back(atoi(numbers.substr(j, i + 1).c_str()));
    vector<int> ans;
    for (int i = 0; i < ins.size(); i++)
      if (ins[i] >= low && ins[i] <= high)
        ans.push_back(ins[i]);
    return ans;
  }
};
